client/src/views/files/handler/FileHandler.vue

```diff
@@ -61,28 +61,26 @@
                 {{ $msTranslate('fileViewers.download') }}
               </ion-button>
               <ion-button
-                class="file-viewer-topbar-buttons__item"
+                class="file-handler-topbar-buttons__item"
                 @click="openWithSystem(contentInfo.path)"
                 v-show="isDesktop() && !atDateTime"
               >
                 <ion-icon :icon="open" />
                 {{ $msTranslate('fileViewers.openWithDefault') }}
               </ion-button>
               <ion-button
-                class="file-viewer-topbar-buttons__item toggle-menu"
-                @click="toggleHeader"
+                class="file-handler-topbar-buttons__item toggle-menu"
+                @click="toggleMainHeader"
+                :class="{ 'header-visible': isHeaderVisible() }"
               >
                 <ion-icon :icon="isHeaderVisible() ? chevronUp : chevronDown" />
                 {{ $msTranslate(isHeaderVisible() ? 'fileViewers.hideMenu' : 'fileViewers.showMenu') }}
               </ion-button>
             </ion-buttons>
           </div>
 
-          <!-- file-viewer component -->
-          <component
-            :is="viewerComponent"
-            :content-info="contentInfo"
-          />
+          <!-- file-handler sub-component -->
+          <slot @handlerReady="handlerReady = true" />
```

Comment:
This seems very weird.

If I understand correctly, you have a component that emits a signal to the parent that emits a signal to its child. I don't know if it's technically wrong but on a design level, it probably is. A parent should pass props to a child, a child should emit signals to the parent.

Reply:
Using a prop to pass the data from the parent to the child, and an event from the child to the parent if needed.